src/testing.h

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#ifndef TESTING_H
+#define TESTING_H
+
+#include "clock.h"
+
+/**
+ * @brief The signature of a test function
+ * @details A graphics test is a function that represents a test of one graphics feature.
+ * It should perform draw calls to the screen that attempt to cover the different use
+ * cases of that feature.
+ * @return A value indicating whether the test was successful or failed
+ */
+typedef int (*test_func_t)(void);
+
+struct test_unit {
+    const char *title;
+    test_func_t test;
+    clock_t runtime_bound;
+    // TODO: NUN-14 Add type or grouping so we can run specific groups of tests
+};
+
+// The list of all tests to be run when the graphics test suite is called upon
+static struct test_unit tests[] = {
```

Comment:
shouldn't this be in the .c file so that tests aren't exposed for other .c files? also, I thought static (not static inline) was bad for .h files since each file that includes the .h will have its own copy.